packages/i18next/src/lib/functions.ts

```diff
@@ -150,11 +162,60 @@ export async function replyLocalized<TKeys extends TFunctionKeys = string, TInte
 	target: Message,
 	options: LocalizedMessageOptions<TKeys, TInterpolationMap>
 ): Promise<Message>;
+/**
+ * Replies to the interaction using the language `keys` from your i18next language setup.
+ * @since 2.4.0
+ * @param target The interaction to reply to.
+ * @param keys The language keys to be sent.
+ * @example
+ * ```typescript
+ * // Using an object to specify the key to send
+ * await replyLocalized(interaction, 'commands/ping:loading');
+ * // ➡ "Pinging..."
+ * ```
+ */
+export async function replyLocalized<TKeys extends TFunctionKeys = string>(
+	target: BaseCommandInteraction | MessageComponentInteraction,
+	keys: TKeys | TKeys[]
+): Promise<ReturnType<(BaseCommandInteraction | MessageComponentInteraction)['reply']>>;
+/**
+ * Replies to the interaction using an object of {@link LocalizedInteractionReplyOptions}.
+ * @since 2.4.0
+ * @param target The interaction to reply to.
+ * @param options The options to be sent, requiring at least `keys` to be passed.
+ * @example
+ * ```typescript
+ * // Using an object to specify the key to send
+ * await replyLocalized(interaction, { keys: 'commands/ping:loading' });
+ * // ➡ "Pinging..."
+ * ```
+ * @example
+ * ```typescript
+ * // Passing interpolation options into i18next
+ * const latency = 42;
+ *
+ * await replyLocalized(interaction, {
+ * 	keys: 'commands/ping:loading',
+ * 	formatOptions: { latency }
+ * });
+ * // ➡ "Pinging... current latency is 42ms."
+ * ```
+ */
 export async function replyLocalized<TKeys extends TFunctionKeys = string, TInterpolationMap extends NonNullObject = StringMap>(
-	target: Message,
-	options: TKeys | TKeys[] | LocalizedMessageOptions<TKeys, TInterpolationMap>
-): Promise<Message> {
-	return target.reply(await resolveOverloads(target, options));
+	target: BaseCommandInteraction | MessageComponentInteraction,
+	options: LocalizedInteractionReplyOptions<TKeys, TInterpolationMap>
+): Promise<ReturnType<(BaseCommandInteraction | MessageComponentInteraction)['reply']>>;
+export async function replyLocalized<TKeys extends TFunctionKeys = string, TInterpolationMap extends NonNullObject = StringMap>(
+	target: BaseCommandInteraction | Message | MessageComponentInteraction,
+	options: TKeys | TKeys[] | LocalizedMessageOptions<TKeys, TInterpolationMap> | LocalizedInteractionReplyOptions<TKeys, TInterpolationMap>
+): Promise<ReturnType<(BaseCommandInteraction | MessageComponentInteraction)['reply']> | Message> {
+	// Handle Interactions:
+	if (target instanceof BaseCommandInteraction || target instanceof MessageComponentInteraction) {
+		return target.reply(await resolveOverloads(target, options))!;
+	}
+
+	// Handle Message:
+	return target.reply(await resolveOverloads(target, options))!;
```

Comment:
This is the exact same code

Reply:
About [this changes request](https://github.com/sapphiredev/plugins/pull/272/files/3d59b5b7208d3d7a485f7c0b385d83c3228d3da2#r851659889)?

Reply:
Plese check the code I wrote the review for...

Reply:
oh sorry